base/applications/regedit/framewnd.c

```diff
@@ -807,6 +814,98 @@ static void ChooseFavorite(LPCWSTR pszFavorite)
+    subkeylen = lstrlenW(subkey);
+    buffer = (WCHAR*)GlobalAlloc(GMEM_FIXED, (rootlen + subkeylen + 1) * sizeof(WCHAR));
+    if (buffer)
+    {
+        lstrcpyW(buffer, rootname);
+        buffer[rootlen - 1] = '\\';
+        lstrcpyW(buffer + rootlen, subkey);
+    }
+    return buffer;
+}
+
+static INT_PTR CALLBACK AddToFavoritesDlgProc(HWND hWnd, UINT uMsg, WPARAM wParam, LPARAM lParam)
+{
+    HWND hName = GetDlgItem(hWnd, IDC_FAVORITENAME);
+    WCHAR name[MAX_PATH];
+    switch (uMsg)
+    {
+        case WM_INITDIALOG:
+        {
+            TVITEM tvi;
+            tvi.mask = TVIF_HANDLE | TVIF_TEXT;
+            tvi.hItem = TreeView_GetSelection(g_pChildWnd->hTreeWnd);
+            tvi.pszText = name;
+            tvi.cchTextMax = _countof(name);
+            if (!TreeView_GetItem(g_pChildWnd->hTreeWnd, &tvi))
+                tvi.pszText[0] = UNICODE_NULL;
+            SendMessageW(hName, WM_SETTEXT, 0, (LPARAM)tvi.pszText);
+            SendMessageW(hName, EM_LIMITTEXT, _countof(name) - 1, 0);
+            return TRUE;
+        }
+        case WM_COMMAND:
+            switch (LOWORD(wParam))
+            {
+                case IDOK:
+                    {
+                        LPCWSTR path;
+                        HKEY hKey;
+                        DWORD err;
+                        if (GetWindowTextW(hName, name, _countof(name)))
+                            err = ERROR_SUCCESS;
+                        else
+                            err = GetLastError();
+
+                        if (err == ERROR_SUCCESS)
+                        {
+                            path = GetItemFullPath(NULL);
+                            if (!path)
+                                err = ERROR_NOT_ENOUGH_MEMORY;
+                        }
+
+                        if (err == ERROR_SUCCESS)
+                            err = RegCreateKeyExW(HKEY_CURRENT_USER, s_szFavoritesRegKey, 0,
+                                                  NULL, 0, KEY_SET_VALUE, NULL, &hKey, NULL);
+                        if (err == ERROR_SUCCESS)
+                        {
+                            err = RegSetValueExW(hKey, name, 0, REG_SZ, (BYTE*)path, (lstrlenW(path) + 1) * sizeof(WCHAR));
+                            RegCloseKey(hKey);
+                        }
+                        if (err)
+                        {
+                            if (!FormatMessageW(FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS,
+                                                NULL, err, 0, name, _countof(name), NULL))
+                                *(UINT*)name = L'?';
+                            MessageBoxW(hWnd, name, NULL, MB_ICONSTOP);
+                        }
+                        if (path)
+                            GlobalFree((HGLOBAL)path);
+                        return EndDialog(hWnd, err);
+                    }
+                case IDCANCEL:
+                    return EndDialog(hWnd, ERROR_CANCELLED);
```

Comment:
Not really useful to return a Win32 error code there, and in the IDOK case above (caller doesn't use that code at all).

Reply:
I know it is not super useful but it costs me nothing so why not?

Reply:
This is in regards to the fact the returned result of the DialogBoxW() call below won't really be used.

Reply:
I know the caller does not care at the moment but I have already calculated the error value, it costs me nothing to return it.